src/BccCode.Linq/ApiClient/ResultList.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System.Collections;
+using System.ComponentModel.DataAnnotations;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+
+namespace BccCode.ApiClient;
+
+[DataContract(Name = "ResultList")]
+public abstract class ResultList
+{
+    protected ResultList(Metadata? meta = null)
+    {
+    }
+
+    /// <summary>
+    /// Gets or Sets Meta
+    /// </summary>
+    [DataMember(Name = "meta", EmitDefaultValue = false)]
+    public Metadata? Meta { get; set; }
+
+    [JsonIgnore]
+    public IEnumerable<object>? Data { get; set; }
+
+    public abstract void AddData(IEnumerable? data);
+
+    [Obsolete($"Use property {nameof(Data)} instead")]
+    public virtual IEnumerable<object> GetData() => Data ?? new List<object>();
+
+    [Obsolete($"Use property {nameof(Data)}.Count() instead")]
+    public virtual long GetCount() => Data?.LongCount() ?? 0;
+}
+
+
+public class ResultList<T> : ResultList, IResultList<T>, IEquatable<ResultList<T>>, IValidatableObject
+{
+    public ResultList()
+    {
+    }
+
+    public ResultList(ResultList<T> resultList)
+    {
+        Data = resultList.Data;
+        Meta = resultList.Meta;
+    }
+
+    public ResultList(IResultList<T> resultList)
+    {
+        if (resultList.Data is List<T> dataList)
+        {
+            Data = dataList;
```

Comment:
IEnumerable<object> isn't the same as IEnumerable<T>... ref comments above.

Reply:
I don't know if I understand you here. The check with ` is List<T>` is done to avoid cloning the list object when not necessary. That is something C# core libraries do as well in their internal code.

Reply:
I think this comment somehow ended up out of context...